engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1684,21 +1741,28 @@ def translateInvalid(invalid: AST.Invalid): Expression = {
     * Currently this only supports documentation comments, and not standarc
     * types of comments as they can't currently be represented.
     *
-    * @param comment the comment to transform
+    * @param tree the comment to transform
     * @return the [[IR]] representation of `comment`
-
-  def translateComment(comment: AST): Comment = {
-    comment match {
-      case AST.Comment(lines) =>
-        Comment.Documentation(
-          lines.mkString("\n"),
-          getIdentifiedLocation(comment)
-        )
-      case _ =>
-        throw new UnhandledEntity(comment, "processComment")
-    }
+    */
+  IR.Comment translateComment(Tree tree) {
+    return switch (tree) {
+      case Tree.ArgumentBlockApplication comment -> {
+        var doc = new StringBuilder();
+        var sep = "";
+        for (var l : comment.getArguments()) {
+          if (l.getExpression() != null) {
+            doc.append(sep);
+            doc.append(l.getExpression().codeRepr());
+            sep = "\n";
+          }
+        }
+        yield new IR$Comment$Documentation(doc.toString(), getIdentifiedLocation(comment), meta(), diag());
```

Comment:
Not a comment, but in some cases comments produce a parse containing `ArgumentBlockApplication`s, so I see how this happened. All source code, including non-semantic text like whitespace and non-doc-comments, has to be represented in the `Tree` types. `ArgumentBlockApplication` is the default block type--any increase in indentation that doesn't fit the rules for an operator block or a body block is an argument block; so to represent some source, you'll get an `ArgumentBlockApplication` that doesn't contain any expressions. This is only the way that plain comments are represented. As I've mentioned, the parser doesn't yet handle doc comments; it treats all comments as plain comments. Doc comments, once implemented, will have their own node and token types.

I've postponed the doc comment implementation (https://www.pivotaltracker.com/story/show/182497173 and https://www.pivotaltracker.com/story/show/182497187) until after integration, because the plan is to lex/parse markdown, and that is not needed for compatibility with the current parser. However, I missed an intermediate step: For compatibility with the current parser, we do need to distinguish doc comments and place them in the AST. I'll implement that soon: https://www.pivotaltracker.com/story/show/183417662. Until then, doc comments don't exist in the lexer, so it's not necessary to try to extract them from the AST. Once they're represented, the data will be much easier to use.

Reply:
Comment processing removed in [466ec48](https://github.com/enso-org/enso/pull/3747/commits/466ec486388ad45adecaad6f6278bf80f84bc821).